sw/device/lib/crypto/impl/hash.c

```diff
@@ -0,0 +1,96 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/crypto/include/hash.h"
+
+#include <stdbool.h>
+
+#include "sw/device/lib/crypto/drivers/kmac.h"
+
+/**
+ * Return the digest size (in bytes) for given hashing mode.
+ *
+ * @param hash_mode Hashing mode (e.g. kHashModeSha256).
+ * @param digest_len Result digest size in bytes.
+ * @return Error status.
+ */
+OT_WARN_UNUSED_RESULT
+static crypto_status_t get_digest_size(hash_mode_t hash_mode,
+                                       size_t *digest_len) {
+  if (digest_len == NULL) {
+    return kCryptoStatusBadArgs;
+  }
+
+  // Below `digest_len` is in bytes, therefore magic values are obtained
+  // after division by 8.
+  switch (hash_mode) {
+    case kHashModeSha3_224:
+      *digest_len = 224 / 8;
+      break;
+    case kHashModeSha256:
+    case kHashModeSha3_256:
```

Comment:
nit: Do we want to keep both names? If so can we add `OT_FALLTHROUGH_INTENDED`?

Reply:
This is intentional (`kHashModeSha256` here refers to SHA-2-256 and `kHashModeSha3_256` is SHA-3-256). I didn't know about `OT_FALLTHROUGH_INTENDED`. Thanks @alphan. I will fix this.